autoextract_poet/utils.py

```diff
@@ -0,0 +1,22 @@
+from autoextract.aio import request_raw
+from autocrawl.autoextract_retry import autoextract_retry
+
+from autoextract_poet.exceptions import QueryLevelError
+
+
+async def request(url: str, page_type: str, full_html=True):
+    query = [dict(url=url, pageType=page_type, fullHtml=full_html)]
+
+    async def _request():
+        response_per_url = await request_raw(query)
+
+        # A single url send, so a single element in the response
+        response = response_per_url[0]
+
+        if "error" in response:
+            raise QueryLevelError(query, response)
+
+        return response
+
+    # FIXME: move retry logic to this lib
```

Comment:
This library is in a private repository. We need to move this code to this pull request and make sure it's using our implementation of the `QueryLevelError` exception.

Reply:
Yes, the retry logic and the exception, etc, should be part of the public repository.

Reply:
I've moved the code from the private repository to this pull request.